app/controllers/groups_controller.rb

```diff
@@ -136,7 +136,7 @@ def assign_scans
       return
     end
     names = next_grouping.non_rejected_student_memberships.map do |u|
-      "#{u.user.first_name} #{u.user.last_name}"
+      u.user.display_name.to_s
```

Comment:
You don't need the `to_s` here, since the display name is already a string. (This applies for all changes below.)